changeset:   4866:18272d4bb51d
user:        nyamatongwe <nyamatongwe@gmail.com>
date:        Sun Nov 03 14:48:22 2013 +1100
summary:     Bug [#1544]. Mac font quality not applied to margin.

diff --git cocoa/ScintillaCocoa.mm cocoa/ScintillaCocoa.mm
--- cocoa/ScintillaCocoa.mm
+++ cocoa/ScintillaCocoa.mm
@@ -1516,16 +1516,26 @@ void ScintillaCocoa::PaintMargin(NSRect 
 {
   CGContextRef gc = (CGContextRef) [[NSGraphicsContext currentContext] graphicsPort];
 
   PRectangle rc = NSRectToPRectangle(aRect);
   rcPaint = rc;
   Surface *sw = Surface::Allocate(SC_TECHNOLOGY_DEFAULT);
   if (sw)
   {
+    CGContextSetAllowsAntialiasing(gc,
+                                   vs.extraFontFlag != SC_EFF_QUALITY_NON_ANTIALIASED);
+    CGContextSetAllowsFontSmoothing(gc,
+                                    vs.extraFontFlag == SC_EFF_QUALITY_LCD_OPTIMIZED);
+#if MAC_OS_X_VERSION_MAX_ALLOWED > MAC_OS_X_VERSION_10_5
+    if (CGContextSetAllowsFontSubpixelPositioning != NULL)
+      CGContextSetAllowsFontSubpixelPositioning(gc,
+						vs.extraFontFlag == SC_EFF_QUALITY_DEFAULT ||
+						vs.extraFontFlag == SC_EFF_QUALITY_LCD_OPTIMIZED);
+#endif
     sw->Init(gc, wMargin.GetID());
     PaintSelMargin(sw, rc);
     sw->Release();
     delete sw;
   }
 }
 
 //--------------------------------------------------------------------------------------------------
